x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AddEditFlyout/index.tsx

```diff
@@ -59,6 +60,9 @@ export function AddEditFlyout({
 
   const callApmApiFromHook = useCallApmApi();
 
+  // get a telemetry UI event tracker
+  const trackApmEvent = useUiTracker({ app: 'apm' });
```

Comment:
I'm not sure the DX here is great. I'm already seeing `useUiTracker({ app: 'apm' })` in multiple places, and I can imagine we'll have to added everywhere we want to track events.

I'd prefer that we either kept it as an inline argument to track event:
```
trackEvent({ app: 'apm', name: 'save_agent_configuration' })
```

Or handle this via context. Then subsequent `trackEvent` hooks will automatically track with `app: apm`.

Reply:
Makes sense 👍 As this is related to DX and is working-but-not-perfect would you mind if we deferred this to the list of followups for all of this work? This would also allow us to gather feedback from all of Observability as these are shared hooks.

Reply:
Sure, that's sounds like good approach 👍